### ISSUE TITLE
Some issues about reproducing the results of the original SBERT paper

### ISSUE THREAD

**Issue by yym6472** [UKPLab/sentence-transformers#597]
Hi, I am recently trying to reproduce the results of SBERT in STS dataset (BERT-base, fine-tune on NLI). However, the results seem lower than those in the original SBERT paper, especially in STS13 and STS14:
- STS13: we got 71.64 while the paper reported 76.53
- STS14: we got 70.49 while the paper reported 73.19

I wonder if my experimental setup is wrong, and want to check the following:
1. In the SBERT paper, how did you calculate the spearman correlation? For STS12-16, there are smaller split in each (e.g. HDL, FNWN, OnWN for STS13). Did you mix them up and calculate spearman together, or individually calculate spearman for each split and report the mean/weighted-mean value? The later is used by SentEval tool, but produces numerically lower results.

2. It seems that the statistics of STS12-16 is not provided in the paper. We obtain the dataset through SentEval, and here are the statistics of datasets:

    | Dataset  | STS12 | STS13 | STS14 | STS15 | STS16 |
    |----------|-------|-------|-------|-------|-------|
    | Number of samples | 3108  | 1500  | 3750  | 3000  | 1186  |

    I wonder if it is consistent with the paper. Specially, in STS13, we notice that one split SMT is missing (the README file says "due to licensing issues"). Will that be the reason why our reproducing results are lower?

**Comment by nreimers** [comment]
Hi @yym6472 
Are you using the pre-trained models or do you train your own models? 

For the paper we used the old pytorch-bert (or pytorch-transformers) package. When it was updated to a more recent version (and re-named to transformers), the performance of the same training scripts dropped. The reason is rather unclear and quite a lot of people were affected by this.

But when loading the pre-trained models, you still got the old results.


In our paper we follow the main scoring way from SemEval. See section 4.1:
https://www.aclweb.org/anthology/S12-1051/

We combine for example for STS2012 all five datasets and compute Spearman correlation over the joined set.

SentEval uses a different setup: It computes the correlation individually for the 5 sets and then averages the five correlation scores.

We recently ran both setups and got the following scores with the pre-trained models for STS12:
bert-base-nli-mean-tokens: 70.97 (all) vs 63.80 (mean of the five splits)
distilroberta-base-paraphrase-v1: 69.16 (all) vs. 69.49 (mean of the five splits)

When combining all datasets, bert-base-nli-mean-tokens is better than distilroberta-base-paraphrase-v1. When computing scores on the individual datasets, distilroberta-base-paraphrase-v1 is much better than bert-base-nli-mean-tokens.

For the paper I also used SentEval, combined all pairs together (as suggested by SemEval 12 paper) and computed Spearman correlation  between predicted scores and gold scores.

**Comment by yym6472** [comment]
@nreimers Thank you so much for your detailed response! I trained the model on my own, using the `training_nil.py` script. I have tried the pre-trained model `bert-base-nli-mean-tokens`, and it works fine and produces the same results as the paper. Also, we find the random seed has a big impact (maybe the STS dataset is small). In another run, the results are much closer to the paper.

**Comment by fiego** [comment]
> Hi, I am recently trying to reproduce the results of SBERT in STS dataset (BERT-base, fine-tune on NLI). However, the results seem lower than those in the original SBERT paper, especially in STS13 and STS14:
> 
> * STS13: we got 71.64 while the paper reported 76.53
> * STS14: we got 70.49 while the paper reported 73.19
> 
> I wonder if my experimental setup is wrong, and want to check the following:
> 
> 1. In the SBERT paper, how did you calculate the spearman correlation? For STS12-16, there are smaller split in each (e.g. HDL, FNWN, OnWN for STS13). Did you mix them up and calculate spearman together, or individually calculate spearman for each split and report the mean/weighted-mean value? The later is used by SentEval tool, but produces numerically lower results.
> 2. It seems that the statistics of STS12-16 is not provided in the paper. We obtain the dataset through SentEval, and here are the statistics of datasets:
>    
>    
>    
>    Dataset
>    STS12
>    STS13
>    STS14
>    STS15
>    STS16
>    
>    
>    
>    
>    Number of samples
>    3108
>    1500
>    3750
>    3000
>    1186
>    
>    
>    
>    I wonder if it is consistent with the paper. Specially, in STS13, we notice that one split SMT is missing (the README file says "due to licensing issues"). Will that be the reason why our reproducing results are lower?

Hi, hello, can you provide the processed data, please? 
The data I downloaded from the Internet doesn't match what you have here.
